test_design.py

```diff
@@ -20,6 +20,22 @@
 from skglm.solvers import cd_solver_path, bcd_solver_path
 
 
+def jit_closure():
+    compiled_classes = dict()
```

Comment:
I assume you can't use `lru_cache` from `functools` because spec is unhashable, right?

Reply:
I forgot about lru_cache ! Did you test it here ? We can make `spec` a tuple if this helps

Reply:
I tried it yesterday but didn't push because of the unhashable issue. If specs can be a tuple, it's more Pythonic :) I'm having a look